### DISCUSSION TITLE
Would data from the Logitech Harmony API be helpful for development?

### DISCUSSION THREAD

**Discussion by TheFiZi** [OMOTE-Community/OMOTE-Firmware#12] [Ideas]
I'm a long time Harmony user and this remote looks like an awesome alternative for when my Harmony inevitably stops working.

I saw in the write up that device support is currently limited and you're hoping to develop an application with, I assume, similar functionality to how the Logitech Harmony App works for programming a remote.

I did some digging today in the MyHarmony App for iOS and was able to pull some data from the Logitech API by searching for a random device, a Denon AVR-A1, just to see what I could get.

Attached is the JSON output:

[jsonExport1.txt](https://github.com/CoretechR/OMOTE/files/12065128/jsonExport1.txt)
[jsonExport2.txt](https://github.com/CoretechR/OMOTE/files/12065129/jsonExport2.txt)

I did a quick skim of the data and I can't see anything that jumps out to me as IR codes that could be helpful for programming a OMOTE but I'm also not an expert on this.

I guess my question is: Is this data at all useful? and would you like more if I could figure out how to get more?

**Comment by CoretechR** [comment]
You're right, It does not seem like there are actual IR codes inside these files. It seems like those codes are downloaded from the Logitech servers to the remote itself.
But thank you anyway!

**Comment by TheFiZi** [comment]
I will see if I can find out what the remote itself downloads when a new device is added. Might take me a bit to figure out how to intercept that traffic.

**Comment by TheFiZi** [comment]
Long time no see, is this useful? In theory it's for the `0` key on the remote for a `Sony STR-DE5*`, I think

```json
 {
  "CommandTypeId": "Number0",
  "DateTaught": null,
  "FunctionGroupId": 2,
  "FunctionId": {
   "IsPersisted": false,
   "Value": 19
  },
  "Id": {
   "IsPersisted": false,
   "Value": 2086
  },
  "IsLearned": false,
  "KeyCode": "G:Sony 12 Bit:()(0x916)():3",
  "Name": "0",
  "Parameters": null,
  "ProtocolId": 1,
  "Raw": null,
  "TransportType": 1
 },
 ```
